### ISSUE TITLE
Pulls from quay.io don't show progress appropriately.

### ISSUE THREAD

**Issue by TomSweeneyRedHat** [containers/buildah#1270]
**Description**
When pulling from an image from quay.io, the progress bar is not shown appropriately.  The image apparently is pulled over OK though.

**Steps to reproduce the issue:**
1.  Push an image to quay.io
2.  Pull the image from quay.io
3.

**Describe the results you received:**
```
# buildah pull quay.io/tomsweeneyredhat/testing:third
Getting image source signatures
Copying blob 8e3ba11ec2a2: 2.10 MiB / ? [--------------------------------=-] 1s 
Copying blob a3ed95caeb02: 32 B / ? [------=-------------------------------] 1s 
Copying blob 4f4fb700ef54: 32 B / ? [------=-------------------------------] 1s 
Copying blob 4f4fb700ef54: 32 B / ? [------=-------------------------------] 1s 
Writing manifest to image destination
Storing signatures
ee052b919fe29862f33a47b0ef3780fafe3e3ae810228c7149997b805d48b99e
```

**Describe the results you expected:**
```
# buildah pull busybox
Getting image source signatures
Copying blob 57c14dd66db0: 738.01 KiB / 738.01 KiB [========================] 0s
Copying config 3a093384ac30: 1.46 KiB / 1.46 KiB [==========================] 0s
Writing manifest to image destination
Storing signatures
3a093384ac306cbac30b67f1585e12b30ab1a899374dabc3170b9bca246f1444
111

**Output of `buildah version`:**

```
# buildah version
Version:         1.6-dev
Go Version:      go1.10.4
Image Spec:      1.0.0
Runtime Spec:    1.0.0
CNI Spec:        0.4.0
libcni Version:  v0.7.0-alpha1
Git Commit:      e7ca330f
Built:           Wed Jan  9 10:49:42 2019
OS/Arch:         linux/amd64

```


**Output of `cat /etc/*release`:**

```
# cat /etc/*release
Fedora release 28 (Twenty Eight)
NAME=Fedora
VERSION="28 (Server Edition)"
ID=fedora
VERSION_ID=28
PLATFORM_ID="platform:f28"
PRETTY_NAME="Fedora 28 (Server Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:28"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=28
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=28
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Server Edition"
VARIANT_ID=server
Fedora release 28 (Twenty Eight)
Fedora release 28 (Twenty Eight)

```

**Output of `uname -a`:**

```
# uname -a
Linux localhost.localdomain 4.18.17-200.fc28.x86_64 #1 SMP Mon Nov 5 18:04:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# cat /etc/containers/storage.conf
# This file is is the configuration file for all tools
# that use the containers/storage library.
# See man 5 containers-storage.conf for more information
# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# Path to an helper program to use for mounting the file system instead of mounting it
# directly.
#mount_program = "/usr/bin/fuse-overlayfs"

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"

# mountopt specifies comma separated list of extra mount options
#mountopt = "nodev"

# Remap-UIDs/GIDs is the mapping from UIDs/GIDs as they should appear inside of
# a container, to UIDs/GIDs as they should appear outside of the container, and
# the length of the range of UIDs/GIDs.  Additional mapped sets can be listed
# and will be heeded by libraries, but there are limits to the number of
# mappings which the kernel will allow when you later attempt to run a
# container.
#
# remap-uids = 0:1668442479:65536
# remap-gids = 0:1668442479:65536

# Remap-User/Group is a name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid or /etc/subgid file.  Mappings are set up starting
# with an in-container ID of 0 and the a host-level ID taken from the lowest
# range that matches the specified name, and using the length of that range.
# Additional ranges are then assigned, using the ranges which specify the
# lowest host-level IDs first, to the lowest not-yet-mapped container-level ID,
# until all of the entries have been used for maps.
#
# remap-user = "storage"
# remap-group = "storage"

[storage.options.thinpool]
# Storage Options for thinpool

# autoextend_percent determines the amount by which pool needs to be
# grown. This is specified in terms of % of pool size. So a value of 20 means
# that when threshold is hit, pool will be grown by 20% of existing
# pool size.
# autoextend_percent = "20"

# autoextend_threshold determines the pool extension threshold in terms
# of percentage of pool size. For example, if threshold is 60, that means when
# pool is 60% full, threshold has been hit.
# autoextend_threshold = "80"

# basesize specifies the size to use when creating the base device, which
# limits the size of images and containers.
# basesize = "10G"

# blocksize specifies a custom blocksize to use for the thin pool.
# blocksize="64k"

# directlvm_device specifies a custom block storage device to use for the
# thin pool. Required if you setup devicemapper.
# directlvm_device = ""

# directlvm_device_force wipes device even if device already has a filesystem.
# directlvm_device_force = "True"

# fs specifies the filesystem type to use for the base device.
# fs="xfs"

# log_level sets the log level of devicemapper.
# 0: LogLevelSuppress 0 (Default)
# 2: LogLevelFatal
# 3: LogLevelErr
# 4: LogLevelWarn
# 5: LogLevelNotice
# 6: LogLevelInfo
# 7: LogLevelDebug
# log_level = "7"

# min_free_space specifies the min free space percent in a thin pool require for
# new device creation to succeed. Valid values are from 0% - 99%.
# Value 0% disables
# min_free_space = "10%"

# mkfsarg specifies extra mkfs arguments to be used when creating the base.
# device.
# mkfsarg = ""

# use_deferred_removal marks devicemapper block device for deferred removal.
# If the thinpool is in use when the driver attempts to remove it, the driver 
# tells the kernel to remove it as soon as possible. Note this does not free
# up the disk space, use deferred deletion to fully remove the thinpool.
# use_deferred_removal = "True"

# use_deferred_deletion marks thinpool device for deferred deletion.
# If the device is busy when the driver attempts to delete it, the driver
# will attempt to delete device every 30 seconds until successful.
# If the program using the driver exits, the driver will continue attempting
# to cleanup the next time the driver is used. Deferred deletion permanently
# deletes the device and all data stored in device will be lost.
# use_deferred_deletion = "True"

# xfs_nospace_max_retries specifies the maximum number of retries XFS should
# attempt to complete IO when ENOSPC (no space) error is returned by
# underlying storage device.
# xfs_nospace_max_retries = "0"

# If specified, use OSTree to deduplicate files with the overlay backend
ostree_repo = ""

# Set to skip a PRIVATE bind mount on the storage home directory.  Only supported by
# certain container storage drivers
skip_mount_home = "false"
```

**Comment by vrothberg** [comment]
@TomSweeneyRedHat this looks like a c/image bug. @mtrmac, can you have a look? I am still on the apparmor fix for Podman but may find some time tomorrow.

**Comment by mtrmac** [comment]
Isn’t quay.io still schema1-only? schema1 manifests don’t provide the size of the layers.

**Comment by TomSweeneyRedHat** [comment]
From what I can see, quay.io is still schema 1.  Perhaps I should add a note to the known issues about this.

**Comment by TomSweeneyRedHat** [comment]
Going to close this one, thanks @mtrmac and @vrothberg for the the attention.

**Comment by vrothberg** [comment]
I didn't know that but I am happy it's not us :)

**Comment by rhatdan** [comment]
Seems weird that we would have the same line printing twice?

**Comment by vrothberg** [comment]
> Seems weird that we would have the same line printing twice?

This usually happens when something prints in parallel to the progress bars. Not sure why it happens in this case but I will have a look.

**Comment by mtrmac** [comment]
(We could probably improve the schema1 output  — e.g. when the copy is done, we do know the total size and total time, and the progress bar could be updated with that, as well as show the “100% complete” state. I don’t know how much of this is possible with the current implementation.

OTOH, if it’s only for schema1, it’s not _that_ attractive to invest into that obsolete format.)
